game-core/src/main/java/games/strategy/triplea/delegate/battle/steps/change/RemoveUnprotectedUnits.java

```diff
@@ -51,20 +51,27 @@ public void execute(final ExecutionStack stack, final IDelegateBridge bridge) {
   }
 
   public void removeUnprotectedUnits(final IDelegateBridge bridge, final BattleState.Side side) {
-    if (Properties.getTransportCasualtiesRestricted(battleState.getGameData())) {
-      checkUndefendedTransports(bridge, side);
-      checkUnprotectedUnits(bridge, side);
+    if (!Properties.getTransportCasualtiesRestricted(battleState.getGameData())) {
+      return;
     }
+    // if we are the attacker, we can retreat instead of dying
```

Comment:
Personally I would not have preserved this comment, it makes sense with some context but otherwise is actually not that helpful and is arguably more cheeky than informative. It does leave me with more questions than answers.

Is there any reason otherwise to not combine the two if statements?